src/components/Sidebar/SidebarContext.spec.js

```diff
@@ -45,4 +58,22 @@ describe('SidebarContext', () => {
 
     expect(buttonEl).toHaveAttribute('data-open', 'true');
   });
+
+  it('should be able to consume context, along with useContext hook', () => {
+    const { getByTestId } = render(
+      <SidebarContextProvider>
```

Comment:
Do you still need to wrap the button in a `<ContextProvider />`? I thought you got it straight from the context hook

Reply:
@robinmetral You won't have to do it within the application, as the Provider is wrapped already at the routing level.
But for the test to work you need to have a Provider at some point.

Reply:
The context hook, only makes it so that you don't have to access the consumer and use the render props style. All of that is abstracted within useContext for you.
https://reactjs.org/docs/hooks-reference.html#usecontext

Reply:
Makes sense, thanks for the details!